### PR TITLE
doc: add sections to migration guide

### PR DESCRIPTION
**Pull request by nashif**
Add sections to the migration guide to make it easier to spot and follow
suggestions of specific areas.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by jhedberg — DISMISSED** [review]

File: doc/releases/migration-guide-3.5.rst [hunk 1]
```diff
@@ -267,10 +254,52 @@ Required changes
...
+  This did not match the zephyr dma-controller binding, so this property is now fixed and set
+  in the SOC devicetree definition. Downstream boards should not override this property and
+  instead use the new driver Kconfig
+  :kconfig:option:`CONFIG_DMA_MCUX_LPC_NUMBER_OF_CHANNELS_ALLOCATED`.
+
+* The LPC55XXX series SOC (except LPC55S06) default main clock has been
+  updated to PLL1 source from XTAL32K running at 144MHZ. If the new
+  kconfig option :kconfig:option:`CONFIG_INIT_PLL1`
+  is disabled then the main clock is muxed to FRO_HR as before.
+
```
Comment (fabiobaltieri) [inline doc/releases/migration-guide-3.5.rst:268]: nit: drop one blank line?

File: doc/releases/migration-guide-3.5.rst [hunk 2]
```diff
@@ -19,40 +22,33 @@ Required changes
   detect if you pass the function memory not belonging to the memory blocks in
   the slab.
 
+* :c:macro:`CONTAINER_OF` now performs type checking, this was very commonly
+  misused to obtain user structure from :c:struct:`k_work` pointers without
+  passing from :c:struct:`k_work_delayable`. This would now result in a build
+  error and have to be done correctly using
+  :c:func:`k_work_delayable_from_work`.
+
+Configuration System (Kconfig)
```
Comment (fabiobaltieri) [inline doc/releases/migration-guide-3.5.rst:31]: This sounds a bit unclear, if feels like it should contain changes of the Kconfig system itself, plus many other areas contains config changes

How about having a "Core OS" are for project wide changes including config options like the bootloader one, and moving the nct38xx under "Device Drivers"?

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
@kartben @fabiobaltieri please take another look

**Review by kartben — APPROVED** [review]
Definitely an improvement, thanks @nashif!
Still unsure about the ordering (and potential nesting of some categories that we'd want) --  CAN Controller showing up before devicetree and subsystems definitely feels weird -- but maybe something for another round of update. 

BTW, there is an extra backtick at line 363 that probably felt through the cracks earlier.

**Comment by jhedberg** [comment]
I'll merge this, so that we can work on the other improvements iteratively.

**Pull request merged**